### PR TITLE
feat(i18n): align backend language contract

### LINKED ISSUE #147: feat: i18n - frontend
## 关联
Parent: #146 (feat: i18n support)

## 任务
- [ ] 引入 i18n 框架（react-i18next 或类似方案）
- [ ] 抽取现有中文硬编码文案为语言包
- [ ] 添加英文语言包
- [ ] 实现语言切换入口（设置页 / 顶栏）
- [ ] 语言偏好持久化（localStorage / 用户配置）
- [ ] 日期、数字等格式本地化

## 注意事项
- monorepo 下各 package 的文案统一管理方式需确认
- 考虑懒加载语言包以控制 bundle size

### PR DESCRIPTION
**Pull request by LLwill**
## Summary

Align frontend locale detection, API request language headers, and backend error handling with the backend i18n contract.

## Related Issue

Closes #147

## Changes

- Persist supported locales through localStorage and the i18n_lang cookie, and send Accept-Language from shared API clients.
- Normalize backend error envelopes and map backend i18n error semantics without branching on localized message text.
- Add apiFetch/apiFetchJson coverage for independent fetch call sites and migrate extension sync to the wrapper.
- Apply login and signed-in language preference flows, including best-effort user language sync.
- Keep the long-term i18n/backend integration rules in docs/i18n-agent-guide.md.

## Testing

- [x] Unit tests added/updated
- [x] pnpm exec vitest run packages/dmworkbase/src/Service/__tests__/apiError.test.ts packages/dmworkbase/src/Service/__tests__/apiFetch.test.ts
- [x] pnpm i18n:check
- [x] git diff --check
- [ ] Manually verified

## Checklist

- [ ] I have read CONTRIBUTING.md
- [x] PR description is in English
- [x] Added tests for my changes
- [x] Updated documentation
- [x] Ran pnpm i18n:check or confirmed the change does not affect UI copy
- [x] Followed commit message conventions (Conventional Commits)

### CODE REVIEW

**Review by Jerry-Xin — DISMISSED** [review]
The PR is in scope for octo-web and the implementation aligns with the stated backend i18n contract; I found no blocking issues.

💬 Non-blocking:
- 🟡 Warning: I could not run the focused Vitest command locally because this checkout does not have `vitest` installed (`pnpm exec` returned `Command "vitest" not found`). `git diff --check origin/main...HEAD` passes.
- 🔵 Suggestion: `apiFetchJson()` in `packages/dmworkbase/src/Service/apiFetch.ts:68` assumes every successful non-204 response has JSON. That matches the migrated call sites, but if this becomes a broader shared wrapper, consider handling empty 200/201 responses or invalid JSON with a clearer error.

✅ Highlights:
- `packages/dmworkbase/src/Service/apiError.ts:86` cleanly normalizes both v2 and legacy error envelopes without branching on localized message text.
- `packages/dmworkbase/src/Service/APIClient.ts:61` and `packages/dmworkbase/src/Service/apiFetch.ts:35` consistently add `Accept-Language` without sending backend-only propagation headers.
- `packages/dmworkbase/src/i18n/detectLocale.ts:58` and `packages/dmworkbase/src/i18n/I18nService.ts:112` implement the requested query/cookie/localStorage language persistence flow.
- `packages/dmworklogin/src/loginSession.ts:20` correctly treats an empty backend `language` as “no preference” and ignores unsupported locale values.

File: packages/dmworkbase/src/Service/apiFetch.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  } catch {
+    return { msg: text, status: response.status };
+  }
+}
+
+export async function apiFetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response> {
+  const response = await fetch(input, {
+    ...init,
+    headers: buildHeaders(input, init),
+  });
```
[the remark this answers is not in the document]
Comment (LLwill) [reply packages/dmworkbase/src/Service/apiFetch.ts:55]: Thanks for flagging this. In this PR `apiFetch` is an internal Octo API helper that preserves the `fetch` input shape while adding `Accept-Language` and normalized API errors. The current production call sites pass either relative API paths or URLs derived from `WKApp.apiClient.config.apiURL`, and this helper does not attach auth credentials on its own. Third-party, presigned, user-provided, or otherwise custom URLs should continue to use raw `fetch` or a purpose-specific client as documented in `docs/i18n-agent-guide.md`. I am treating this as a generic-wrapper false positive for this change.

**Review by lml2468 — CHANGES_REQUESTED** [review]
## Review: feat(i18n): align backend language contract

### Summary
Changes requested. The core locale detection, persistence, and normalized error handling are directionally sound, but the PR introduces a credential-leak risk in the new language sync path and does not apply the `Accept-Language` contract to existing custom Axios clients.

### Findings
| # | Severity | File | Line | Issue |
|---|---|---|---|---|
| 1 | P1 | packages/dmworkbase/src/Components/NavRail/NavLanguageSwitcher.tsx | 34 | The failed language-sync path logs the raw rejection object from `updateUserLanguagePreference`. That promise rejects with `APIClientRejectedError`, which includes the raw Axios error; APIClient injects the `token` request header before the request, so the logged object can expose credentials in the browser console or captured client logs. Log only sanitized fields such as status/code/message, or omit the error object entirely. |
| 2 | P1 | packages/dmworksummary/src/api/summaryApi.ts | 26 | `summaryAxios` is an `axios.create()` instance with its own request interceptor, so it does not inherit the new `APIClient` global Axios interceptor that adds `Accept-Language`. The same pattern exists in the matter/todo custom clients. As a result, a meaningful set of internal backend requests still omit `Accept-Language`, leaving the backend language contract only partially implemented. Add `buildAcceptLanguage()` to these custom clients or route them through the shared API wrapper. |

### Verdict
CHANGES_REQUESTED

**Review by lml2468 — DISMISSED** [review]
[APPROVE]

Clean unification of API error handling and backend i18n contract alignment.

## 🔵 Suggestions (non-blocking)

**1. `ConnectionStatus` latency measurement behavior change**
`apiFetch` throws on non-2xx, so `/health` errors now skip latency recording instead of measuring round-trip time regardless. Low risk since health endpoints should return 200, but worth noting.

**2. `InviteLanding.loadInviteInfo` — 5xx maps to "invalid code"**
When the server returns 5xx, `normalizeApiError` sets `backendMessage: undefined` (correctly hiding internals). In the catch block, `hasBackendMessage` is false and `status > 0`, so it falls into `t("app.invite.invalidCode")` instead of a generic error. Consider checking `isInternalApiError` or using the already-normalized `e.message` for all `ApiFetchError` instances.

**3. v2 envelope 404 falls back to "unknown error"**
The legacy path has explicit 404 → `t("base.api.error.notFound")` handling, but the v2 path does not — a v2 404 without `error.message` would show "unknown error". Very low risk since v2 envelopes should always include `message`.

## ✅ Highlights

- Centralized `normalizeApiError` handles both v2 (`error.code` / `error.http_status`) and legacy (`msg` / `status`) envelopes cleanly
- Internal/5xx errors correctly hide raw backend text (SQL, stack traces) from users
- `APIClientRejectedError` is a backward-compatible superset of the old `{ error, msg, status }` shape
- Comprehensive test coverage: `apiError`, `apiFetch`, `APIClient.error`, `NavLanguageSwitcher`, `loginSession` language preference
- Remaining raw `fetch` calls match the documented exceptions (blobs, probes, OIDC, static resources)
- Best-effort backend language sync on NavRail switch with proper fire-and-forget error handling
- Cookie-based locale persistence with `SameSite=Lax` and correct detection priority chain

**Review by lml2468 — APPROVED** [review]
## Re-review: feat(i18n): align backend language contract

### Previous Findings Status
| # | Previous Issue | Status |
|---|---|---|
| 1 | Raw error logged may expose token header credentials | ✅ Fixed — `summarizeLanguageSyncError()` now extracts only status/code/msg; test asserts `secret-token` never appears in logged output |
| 2 | Custom Axios clients miss Accept-Language interceptor | ✅ Fixed — `buildAcceptLanguage()` added to summaryAxios, matterBridge, and todoApi interceptors with test coverage |

### New Changes Assessment
Clean incremental fix. No new issues introduced. Test coverage confirms both fixes work correctly.

### Verdict
APPROVED

**Review by Jerry-Xin — APPROVED** [review]
The PR is in scope for `octo-web` and aligns shared i18n locale detection, API language headers, backend error normalization, and signed-in language sync without introducing blocking issues.

💬 Non-blocking

🟡 Warning: I could not re-run the focused Vitest suite or `pnpm i18n:check` in this checkout because `node_modules` is missing (`vitest` and `typescript` were not available). `git diff --check origin/main...HEAD` passed.

🔵 Suggestion: `apiFetchJson()` assumes every non-204 success response has a JSON body. That is reasonable for current migrated call sites, but if any backend endpoint can return `200` with an empty body, this will now throw instead of behaving like the previous `resp.json().catch(...)` patterns. Consider either documenting that contract or adding a targeted test around empty success bodies. See `packages/dmworkbase/src/Service/apiFetch.ts:68`.

✅ Highlights

- `apiFetch` and `APIClient` now inject `Accept-Language` from the shared locale state without sending trusted backend-only headers. See `packages/dmworkbase/src/Service/apiFetch.ts:30` and `packages/dmworkbase/src/Service/APIClient.ts:61`.
- Backend error normalization uses stable codes/statuses and hides internal/5xx backend text. See `packages/dmworkbase/src/Service/apiError.ts:98` and `packages/dmworkbase/src/Service/apiError.ts:140`.
- Locale persistence and detection now cover query params, `i18n_lang`, localStorage, and browser fallback consistently. See `packages/dmworkbase/src/i18n/detectLocale.ts:58` and `packages/dmworkbase/src/i18n/I18nService.ts:112`.
- Signed-in language switching updates locally first and syncs the user preference best-effort without rolling back local UI state on sync failure. See `packages/dmworkbase/src/Components/NavRail/NavLanguageSwitcher.tsx:40`.
